[CI] XPackRestIT class failing

**Build Scans:**
- [elasticsearch-intake #9755 / rest-compat](https://gradle-enterprise.elastic.co/s/nk35wdht4tkkq)
- [elasticsearch-intake #9755 / part3](https://gradle-enterprise.elastic.co/s/gifja6r7f7vcy)
- [elasticsearch-pull-request #31340 / rest-compatibility](https://gradle-enterprise.elastic.co/s/3tmxstvhyswnw)
- [elasticsearch-pull-request #31281 / part-3](https://gradle-enterprise.elastic.co/s/qzrdidxmoar24)
- [elasticsearch-periodic-platform-support #4039 / oraclelinux-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/yrluj4m6v4rr2)
- [elasticsearch-periodic #4066 / encryption-at-rest](https://gradle-enterprise.elastic.co/s/tnnwhjoval7j4)
- [elasticsearch-periodic-platform-support #4039 / ubuntu-2004_platform-support-unix](https://gradle-enterprise.elastic.co/s/a2w2m453bjv6g)
- [elasticsearch-periodic-platform-support #4039 / rhel-7_platform-support-unix](https://gradle-enterprise.elastic.co/s/6bpttefqlygwk)
- [elasticsearch-periodic-platform-support #4039 / ubuntu-2204_platform-support-unix](https://gradle-enterprise.elastic.co/s/5hcspkzh3gwyy)
- [elasticsearch-periodic-platform-support #4039 / oraclelinux-7_platform-support-unix](https://gradle-enterprise.elastic.co/s/rr4jzlb6wclfe)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:yamlRestCompatTest" --tests "org.elasticsearch.xpack.test.rest.XPackRestIT.test {p0=rollup/rollup_search/Search error against live index}" -Dtests.seed=1158360447451BB -Dtests.locale=es-419 -Dtests.timezone=US/Central -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.test.rest.XPackRestIT'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.test.rest.XPackRestIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [rollup/rollup_search:34]: got unexpected warning header [
	299 Elasticsearch-9.0.0-3d6e9a2850a5d91c7fee0dbe6153903b909c4937 "The rollup functionality will be removed in in Elasticsearch 9.0. See docs for more information."
]

```

**Issue Reasons:**
- [main] 105 failures in class org.elasticsearch.xpack.test.rest.XPackRestIT (10.6% fail rate in 986 executions)
- [main] 3 failures in step rest-compat (6.5% fail rate in 46 executions)
- [main] 5 failures in step part-3 (2.6% fail rate in 196 executions)
- [main] 2 failures in step part3 (4.7% fail rate in 43 executions)
- [main] 9 failures in step rest-compatibility (4.0% fail rate in 227 executions)
- [main] 4 failures in step encryption-at-rest (30.8% fail rate in 13 executions)
- [main] 4 failures in step sles-12_platform-support-unix (33.3% fail rate in 12 executions)
- [main] 4 failures in step ubuntu-2004_platform-support-unix (50.0% fail rate in 8 executions)
- [main] 2 failures in step ubuntu-1804_platform-support-unix (18.2% fail rate in 11 executions)
- [main] 2 failures in step debian-10_platform-support-unix (16.7% fail rate in 12 executions)
- [main] 4 failures in step rhel-7_platform-support-unix (33.3% fail rate in 12 executions)
- [main] 2 failures in step centos-7_platform-support-unix (16.7% fail rate in 12 executions)
- [main] 3 failures in step oraclelinux-8_platform-support-unix (33.3% fail rate in 9 executions)
- [main] 5 failures in step rhel-9_platform-support-unix (41.7% fail rate in 12 executions)
- [main] 3 failures in step ubuntu-2204_platform-support-unix (27.3% fail rate in 11 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -210,8 +210,6 @@ tests:
 - class: org.elasticsearch.xpack.ml.integration.MlJobIT
   method: testDeleteJobAfterMissingAliases
   issue: https://github.com/elastic/elasticsearch/issues/112823
-- class: org.elasticsearch.xpack.test.rest.XPackRestIT
-  issue: https://github.com/elastic/elasticsearch/issues/111944
 - class: org.elasticsearch.datastreams.logsdb.qa.StandardVersusLogsIndexModeRandomDataChallengeRestIT
   method: testDateHistogramAggregation
   issue: https://github.com/elastic/elasticsearch/issues/112919

